contracts/Everlong.sol

```diff
@@ -64,113 +62,16 @@ import { IHyperdrive } from "hyperdrive/contracts/src/interfaces/IHyperdrive.sol
 /// @custom:disclaimer The language used in this code is for coding convenience
 ///                    only, and is not intended to, and does not, have any
 ///                    particular legal or regulatory significance.
-contract Everlong is Admin, ERC4626, Positions {
-    /// @notice Virtual shares are used to mitigate inflation attacks.
-    bool public constant useVirtualShares = true;
-
-    /// @notice Used to reduce the feasibility of an inflation attack.
-    /// TODO: Determine the appropriate value for our case. Current value
-    ///       was picked arbitrarily.
-    uint8 public constant decimalsOffset = 3;
-
-    /// @notice Address of the Hyperdrive instance wrapped by Everlong.
-    address public immutable hyperdrive;
-
-    /// @notice ERC20 token used for deposits, idle liquidity, and
-    ///         the purchase of bonds from the Hyperdrive instance.
-    address internal immutable _asset;
-
-    /// @notice Decimals used by the `_asset`.
-    uint8 internal immutable _decimals;
-
-    /// @notice Name of the Everlong token.
-    string internal _name;
-
-    /// @notice Symbol of the Everlong token.
-    string internal _symbol;
-
+abstract contract Everlong is EverlongAdmin, EverlongPositions {
```

Comment:
*Why is this abstract? *
It shouldn't be :)

*Why use an internal directory?*
I had originally thought Hyperdrive's "internal" directory was named due to it housing various slices of internal state, intent was to do the same here... could also have called it "abstract" as it would house abstract contracts managing (more or less) their own internal bit of functionality.

Do you think changing the directory name to `abstract/` would be better?

Reply:
Nah, `internal` is fine with me. I could also see something like `core` being a good fit. `internal` and `external` are trying to capture whether or not the logic is about the internal mechanics or the external interface, but it's definitely doing double duty since `internal` is all of the core logic.

I will say that if it's left as `internal`, it probably makes sense to leave `Everlong.sol` in the `contracts/` directory since that contract is more "external." You could also move it to `external` if you want. I don't have a strong opinion.